L1Trigger/L1TTrackerDTC/python/Producer_Defaults_cfi.py

```diff
@@ -0,0 +1,62 @@
+    ProductBranch           = cms.string  ( "StubAccepted" ),                                  #
+    DataFormat              = cms.string  ( "Hybrid" ),                                        # hybrid and tmtt format supported
+    OffsetDetIdDSV          = cms.int32   (  1 ),                                              # tk layout det id minus DetSetVec->detId
+    OffsetDetIdTP           = cms.int32   ( -1 )                                               # tk layout det id minus TrackerTopology lower det id
+  ),
+
+  #=== router parameter
+
+  ParamsRouter = cms.PSet (
+    EnableTruncation = cms.bool  ( True  ), # enables emulation of truncation
+    FreqDTC          = cms.double( 360.  ), # Frequency in MHz, has to be integer multiple of FreqLHC
+    TMP_TFP          = cms.int32 (  18   ), # time multiplexed period of track finding processor
+    NumFramesInfra   = cms.int32 (   6   ), # needed gap between events of emp-infrastructure firmware
+    NumRoutingBlocks = cms.int32 (   2   ), # number of systiloic arrays in stub router firmware
+    SizeStack        = cms.int32 (  64   )  # fifo depth in stub router firmware
+  ),
+
+  #=== converter parameter
+
+  ParamsConverter = cms.PSet (
+    WidthRowLUT    = cms.int32 ( 4 ), # number of row bits used in look up table
+    WidthQoverPt   = cms.int32 ( 9 )  # number of bits used for stub qOverPt. lut addr is col + bend = 11 => 1 BRAM -> 18 bits for min and max val -> 9
+  ),
+
+  #=== Tracker parameter
+
+  ParamsTracker           = cms.PSet (
+    NumRegions            = cms.int32 (  9      ), # number of phi slices the outer tracker readout is organized in
+    NumOverlappingRegions = cms.int32 (  2      ), # number of regions a reconstructable particles may cross
+    NumDTCsPerRegion      = cms.int32 ( 24      ), # number of DTC boards used to readout a detector region
+    NumModulesPerDTC      = cms.int32 ( 72      ), # max number of sensor modules connected to one DTC board
+    TMP_FE                = cms.int32 (  8      ), # number of events collected in front-end
+    WidthBend             = cms.int32 (  6      ), # number of bits used for internal stub bend
+    WidthCol              = cms.int32 (  5      ), # number of bits used for internal stub column
+    WidthRow              = cms.int32 ( 11      ), # number of bits used for internal stub row
+    BaseBend              = cms.double(   .25   ), # precision of internal stub bend in pitch units
+    BaseCol               = cms.double(  1.     ), # precision of internal stub column in pitch units
+    BaseRow               = cms.double(   .5    ), # precision of internal stub row in pitch units
+    BendCut               = cms.double(  1.3125 ), # used stub bend uncertainty in pitch units
+    FreqLHC               = cms.double( 40.     )  # LHC bunch crossing rate in MHz
+  ),
+
+  #=== f/w constants
+
+  ParamsFW = cms.PSet (
+    SpeedOfLight = cms.double( 2.99792458       ), # in e8 m/s
```

Comment:
this should be obtained as a constant in the code from CLHEP::c_light

Reply:
see comment on Format_Hybrid_cfi.py

Reply:
c_light is used elsewere in the emulator, e.g. https://github.com/cms-sw/cmssw/blob/master/L1Trigger/TrackTrigger/src/StubPtConsistency.cc
this is a constant of nature and it should be obtained consistently

Reply:
The firmware, written in VHDL, has no access to any c/c++ libraries. The only consistent way I found is the way you see, gathering of hard coded constant in a configuration file in both worlds, f/w and s/w. I am open to better solutions but I can't see them in the moment.

Reply:
To me, the best approach is to use the value from the central source in the emulator, and copy that value into the firmware.

Reply:
I found no way to copy the value to f/w in human readable form while ensuring equality. I would like to avoid specifying machine readable constants, since it makes it harder to compare.